flow/error_definitions.h

```diff
@@ -226,6 +226,7 @@ ERROR( invalid_tenant_name, 2134, "Tenant name cannot begin with \\xff");
 ERROR( tenant_prefix_allocator_conflict, 2135, "The database already has keys stored at the prefix allocated for the tenant");
 ERROR( tenants_disabled, 2136, "Tenants have been disabled in the cluster");
 ERROR( unknown_tenant, 2137, "Tenant is not available from this server")
+ERROR( illegal_tenant_access, 2138, "Tenant information mismatch")
```

Comment:
I think the error name serves as a decent description here

```suggestion
ERROR( illegal_tenant_access, 2138, "Illegal tenant access")
```